assets/js/util/index.js

```diff
@@ -590,30 +591,50 @@ export const findTagInHtmlContent = ( html, module ) => {
  * while requesting list of accounts.
  *
  * @param {string} module Module slug.
+ *
+ * @param {string|null} The tag id if found, otherwise null.
  */
 export const getExistingTag = async ( module ) => {
-	try {
-		let tagFound = data.getCache( module + '::existingTag', 300 );
-
-		if ( 'undefined' === typeof tagFound ) {
-			const html = await fetch( `${ googlesitekit.admin.homeURL }?tagverify=1&timestamp=${ Date.now() }` ).then( ( res ) => {
-				return res.text();
-			} );
-
-			tagFound = findTagInHtmlContent( html, module );
-			if ( ! tagFound ) {
-				tagFound = '';
+	const CACHE_KEY = `${ module }::existingTag`;
+	const { homeURL, ampMode } = googlesitekit.admin;
+
+	let tagFound = data.getCache( CACHE_KEY, 300 );
+
+	if ( ! tagFound ) {
+		try {
+			tagFound = await scrapeTag( addQueryArgs( homeURL, { tagverify: 1, timestamp: Date.now() } ), module );
+
+			if ( ! tagFound && 'secondary' === ampMode ) {
+				tagFound = await apiFetch( { path: '/wp/v2/posts?per_page=1' } ).then(
+					// Scrape the first post in AMP mode, if there is one.
+					( posts ) => posts.slice( 0, 1 ).map( async ( post ) => {
+						return await scrapeTag( addQueryArgs( post.link, { amp: 1 } ), module );
+					} ).pop()
+				);
 			}
-		}
 
-		data.setCache( module + '::existingTag', tagFound );
+			// Only set/renew the cache if a tag was found.
+			data.setCache( CACHE_KEY, tagFound || undefined );
+		} catch ( err ) {}
```

Comment:
Just to clarify, this part can remain as is I believe. This should only be reached if something unexpected happened during tag detection, so we shouldn't reset the cache based on something being broken.

Reply:
I'm not quite sure what you mean. Currently the cache will only be set if there is no error thrown while detecting which is how it was before as well.

Reply:
Right, I was just confirming that this part should not be changed in regards to my other feedback.